src/components/Handlers/withHandlers.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+function getComponentFromHandler(event, stripes, module, data) {
+  const m = module.getModule();
+  const eventHander = m[module.handlerName];
+  if (!eventHander) return null;
+  return eventHander(event, stripes, data);
+}
+
+function getComponentsFromHandlers(event, stripes, modules, data) {
+  return modules.reduce((acc, m) => {
+    const component = getComponentFromHandler(event, stripes, m, data);
+    if (component) {
+      acc.push(stripes.connect(component));
+    }
+    return acc;
+  }, []);
+}
+
+export default function withHandlers(WrappedComponent) {
+  class WithHandlers extends React.Component {
```

Comment:
I find it a bit confusing that you're using the same name for the class and the function that creates it.

Reply:
@MikeTaylor yeah I stole this pattern from a different HOC in `stripes-core`: https://github.com/folio-org/stripes-core/blob/master/src/components/Modules/withModules.js
I wonder perhaps we don't even need a HOC here since all we do is just use these 2 functions.